ledger/src/bank_transaction_executor.rs

```diff
@@ -0,0 +1,547 @@
+        while let Ok((
+            response_sender,
+            BankTransactionExecutionRequest {
+                bank,
+                tx,
+                transaction_status_sender,
+                replay_vote_sender,
+                cost_capacity_meter,
+            },
+        )) = receiver.recv()
+        {
+            let signature = *tx.signature();
+
+            let txs = vec![tx];
+            let mut batch = TransactionBatch::new(vec![Ok(())], &bank, Cow::Owned(txs));
+            batch.set_needs_unlock(false);
+
+            let mut timings = ExecuteTimings::default();
+            let execution_result = execute_batch(
+                &batch,
+                &bank,
+                transaction_status_sender.as_ref(),
+                replay_vote_sender.as_ref(),
+                &mut timings,
+                cost_capacity_meter,
+            );
+
+            if response_sender
+                .send(BankTransactionExecutionResponse {
+                    result: execution_result,
+                    timings,
+                    signature,
+                })
+                .is_err()
+            {
+                warn!("error sending back results");
+            }
+        }
+    }
+}
+
+fn execute_batch(
+    batch: &TransactionBatch,
+    bank: &Arc<Bank>,
+    transaction_status_sender: Option<&TransactionStatusSender>,
+    replay_vote_sender: Option<&ReplayVoteSender>,
+    timings: &mut ExecuteTimings,
+    cost_capacity_meter: Arc<RwLock<BlockCostCapacityMeter>>,
+) -> transaction::Result<()> {
+    let record_token_balances = transaction_status_sender.is_some();
+
+    let mut mint_decimals: HashMap<Pubkey, u8> = HashMap::new();
+
+    let pre_token_balances = if record_token_balances {
+        collect_token_balances(bank, batch, &mut mint_decimals, None)
+    } else {
+        vec![]
+    };
+
+    let pre_process_units: u64 = aggregate_total_execution_units(timings);
+
+    let (tx_results, balances) = batch.bank().load_execute_and_commit_transactions(
+        batch,
+        MAX_PROCESSING_AGE,
+        transaction_status_sender.is_some(),
+        transaction_status_sender.is_some(),
+        transaction_status_sender.is_some(),
+        timings,
+    );
+
+    if bank
+        .feature_set
+        .is_active(&feature_set::gate_large_block::id())
+    {
+        let execution_cost_units = aggregate_total_execution_units(timings) - pre_process_units;
+        let remaining_block_cost_cap = cost_capacity_meter
+            .write()
+            .unwrap()
+            .accumulate(execution_cost_units);
+
+        debug!(
+            "bank {} executed a batch, number of transactions {}, total execute cu {}, remaining block cost cap {}",
+            bank.slot(),
+            batch.sanitized_transactions().len(),
+            execution_cost_units,
+            remaining_block_cost_cap,
+        );
+
+        if remaining_block_cost_cap == 0_u64 {
```

Comment:
```suggestion
        if remaining_block_cost_cap == 0 {
```

Reply:
copied from blockstore_processor.rs, going to leave as is